iib/workers/tasks/utils.py

```diff
@@ -723,11 +723,19 @@ def run_cmd(cmd, params=None, exc_msg=None, strict=True):
     if strict and response.returncode != 0:
         log.error('The command "%s" failed with: %s', ' '.join(cmd), response.stderr)
         if cmd[0] == 'opm':
+            # The new OPM shows the error at the beginning, let's try with this first
+            regex = r'^(?:Error: )(.+)'
+            msg = response.stderr.splitlines()
+            match = re.match(regex, "".join(msg))
```

Comment:
@JAVGan Could you help understand how does this work?

 It looks like it will create list of lines from `stderr`; all new line characters `\n` are removed by [splitlines](https://docs.python.org/3/library/stdtypes.html#str.splitlines) and then the message is joined to one long string. 
It means it returns everything - not only the error message? Or there is nothing else on `stderr` just this error message?

Thanks!

Reply:
Hi @lipoja, you're correct.

The `response.stderr` contains a string with the entire error message from `subprocess.Popen`.

In the original code they were creating a list of strings for each line to search for the error in reversed way for each substring, like this:

```
for msg in reversed(response.stderr.splitlines()):
    match = re.match(regex, msg)
    ...
```

Then I tried to combine the legacy code with my new and moved this command to `msg` in line 728 to use in both regex, but in fact it doesn't need to be like this.

Since I'm splitting and then joining it make no sense to keep this instruction like this.

I can change it to parse the entire string in my regex and let the split just occur in the reverse search.

Reply:
Pushing the amended code